src/App.svelte

```diff
@@ -87,6 +86,22 @@
   });
   page();
 
+  // set up a handler to update our URL when page state changes (we do this here,
+  // instead of the store because we want to wait until we've initialized any
+  // initial state before doing this)
+  $: {
+    // create nice, short urls that don't incorporate falsely values
+    // and convert booleans to integers
+    const query = stringify(
+      mapValues(
+        pickBy($pageState, (v) => (typeof v !== "string" && v) || v.length > 0),
+        (v) => (typeof v === "boolean" ? +v : v)
+      )
+    );
```

Comment:
The comments are definitely helpful, but I think we should split the logic to smaller functions to improve readability (tbh I'm asking for a selfish reason, my brain can't comprehend too many logic circuits at once lol)

Reply:
I think keeping it in one function is for the best, but I'll write a comment to say more about what's going on here.

Reply:
I split out the state handling from the stringify step and added a couple more comments, hopefully that helps.